onnxruntime/core/framework/mem_pattern_planner.h

```diff
@@ -65,7 +65,7 @@ class MemPatternPlanner {
     return false;
   }
 
-  void TraceAllocation(int ml_value_idx, const AllocPlanPerValue::ProgramCounter& counter, size_t size) {
+  void TraceAllocation(int ml_value_idx, const AllocPlanPerValue::ProgramCounter& counter, size_t size, bool insert_end = false) {
```

Comment:
nit. a more readable name for the flag could sth like "disable_memory_time_sharing" ... 

insert_end means the user need to understand the underlaying data structure and mechanism to decide if such flag should be set. usually a good name describes the functionality, and leave out the implementation details.